### PR TITLE
Vagrant and oi-userland integration 

### PR DESCRIPTION
**Pull request by xen0l**
This PR adds some basic information how to use Vagrant with oi-userland. Hopefuly, this will help increase the rate of contributions from new people. It would also be good if people could test this on non-OI platforms.

### CODE REVIEW

File: doc/vagrant.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+* Configure remote repository, so you can fetch updates easily:
+
+  ```
+  git remote add upstream https://github.com/OpenIndiana/oi-userland.git
+  ```
+
+  Updates are fetched with the following commands:
+
+  ```
+  git pull upstream oi/hipster
```
Comment (Toasterson) [inline doc/vagrant.md]: use git pull --rebase. that removes the merge commits and makes it easier to make PR's against oi-userland

File: Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    v.customize ["modifyvm", :id, "--cpus", cpus]
+  end
+
+  # Once vagrant is able to chown files on OpenIndiana, chown line should be
+  # removed from this part.
+  config.vm.provision "shell", inline: <<-SHELL
+    pfexec chown -R vagrant:vagrant /vagrant
+    pfexec pkg install build-essential
+
+    cd /vagrant && gmake setup
```
Comment (Toasterson) [inline Vagrantfile:68]: Be aware that when you have rsync, any changes in the guest will be overwritten every time vagrant does rsync. so the repo will be removed if you not make it on the host.

File: Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  # Disable automatic box update checking. If you disable this, then
+  # boxes will only be checked for updates when the user runs
+  # `vagrant box outdated`. This is not recommended.
+  config.vm.box_check_update = true
+
+  # Unless OpenIndiana is better supported in vagrant, we have to use this
+  # workaround. The problem is with vagrant creating folder as root:root,
+  # but rsync connects as vagrant and fails to write.
+  config.vm.synced_folder ".", "/vagrant", type: "rsync",
+    rsync__rsync_path: "pfexec rsync", owner: "vagrant", group: "vagrant"
```
Comment (Toasterson) [inline Vagrantfile:32]: Does this still execute rsync with --delete ? if so any changes in the vm will be deleted everytime the rsync starts.

File: doc/vagrant.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
+# Using oi-userland with Vagrant
+
+oi-userland includes Vagrantfile, which allows contributing from non-OpenIndiana
+platforms such as Mac OS X, Linux or Windows. Currently, only the VirtualBox
+provider is supported!. Vagrant is suitable for cases when you don't have access to OpenIndiana host and want to develop/contribute to oi-userland.
```
Comment (pyhalov) [inline doc/vagrant.md]: Please, fix "!."

File: Vagrantfile (hunk 3 above)
Comment (xen0l) [inline Vagrantfile:32]: yes, it does. I can change it to remove --delete.

File: Vagrantfile (hunk 2 above)
Comment (xen0l) [inline Vagrantfile:68]: I am aware of that, however there isn't any other neat possibility just to use rsync/rsync-auto. VirtualBox shared folders and other solutions do not work everywhere.

**Pull request merged**